src/brevitas/quant_tensor/groupwise_float_quant_tensor.py

```diff
@@ -0,0 +1,325 @@
+
+    @staticmethod
+    def is_zero_zero_point(tensor):
+        GroupwiseFloatQuantTensor.check_input_type(tensor)
+        return (tensor.zero_point == 0.).all()
+
+    def check_scaling_factors_same(self, other):
+        if self.training:
+            return True
+        if not torch.allclose(self.scale, other.scale):
+            raise RuntimeError("Scaling factors are different")
+
+    def check_zero_points_same(self, other):
+        if self.training:
+            return True
+        if not torch.allclose(self.zero_point, other.zero_point):
+            raise RuntimeError("Zero points are different")
+
+    def check_bit_width_same(self, other):
+        if not torch.allclose(self.exponent_bit_width,
+                              other.exponent_bit_width) and not torch.allclose(
+                                  self.mantissa_bit_width, other.mantissa_bit_width):
+            raise RuntimeError("Bit widths are different")
+
+    def check_exponent_bias(self, other):
+        if not torch.allclose(self.exponent_bias, other.exponent_bias):
+            raise RuntimeError("Bit widths are different")
+
+    def check_inf_nan_same(self, other):
+        if not (set(self.inf_values) == set(other.inf_values)) and not (set(self.nan_values) == set(
+                other.nan_values)):
+            raise RuntimeError("Floating point representations are different")
+
+    def check_sign_same(self, other):
+        if not self.signed == other.signed:
+            raise RuntimeError("Signs are different")
+
+    def view(self, *args, **kwargs):
+        return self.value.view(*args, **kwargs)  #self.set(value=self.value.view(*args, **kwargs))
+
+    def reshape(self, *args, **kwargs):
+        return self.value.reshape(
+            *args, **kwargs)  # self.set(value=self.value.reshape(*args, **kwargs))
+
+    def flatten(self, *args, **kwargs):
+        return self.value.flatten(
+            *args, **kwargs)  #self.set(value=self.value.flatten(*args, **kwargs))
+
+    def transpose(self, *args, **kwargs):
+        value = self.value.transpose(*args, **kwargs)
+        return value
+
+    def permute(self, *args, **kwargs):
+        value = self.value.permute(*args, **kwargs)
+        return value
+
+    # Reference: https://docs.python.org/3/reference/datamodel.html#emulating-numeric-types
+
+    def __neg__(self):
+        neg_deq = -self.minifloat(float_datatype=True)
+        _, scale, zp = self.expand()
+
+        neg_value = (-neg_deq - zp) * scale
+        # In case the dtype of self.minifloat is different from the one of the scale
+        neg_value = neg_value.type(scale.dtype)
+        neg_value = GroupwiseFloatQuantTensor.from_expanded(
+            neg_value, self.group_dim, self.group_size, compress=False)
+        scale = GroupwiseFloatQuantTensor.from_expanded(
+            scale, self.group_dim, self.group_size, compress=True)
+        if self.signed:
+            return GroupwiseFloatQuantTensor(
```

Comment:
nitpicky, but maybe worth putting these in the correct order as well (I know it makes no difference), but the other errors may be due to copy/paste from functions like these...